test/integration/associations/belongs-to-many.test.js

```diff
@@ -556,6 +556,95 @@ describe(Support.getTestDialectTeaser('BelongsToMany'), () => {
         });
       });
     });
+
+    it('answers true for labels that have been assigned multitple times', function() {
+      this.ArticleLabel = this.sequelize.define('ArticleLabel', {
+        id: {
+          primaryKey: true,
+          type: DataTypes.INTEGER,
+          autoIncrement: true
+        },
+        relevance: {
+          type: DataTypes.DECIMAL,
+          validate: {
+            min: 0,
+            max: 1
+          }
+        }
+      });
+      this.Article.belongsToMany(this.Label, { through: { model: this.ArticleLabel, unique: false } });
+      this.Label.belongsToMany(this.Article, { through: { model: this.ArticleLabel, unique: false } });
+
+      return this.sequelize.sync({ force: true }).then(() => {
+        return Promise.all([
```

Comment:
Please avoid deep nesting.

Reply:
Should I be refactoring each of the tests above to match a non-nested style or is it preferable to keep a consistence style throughout the file? With these tests being an iteration of the above, I'd opted to keep it similar but can change it in either direction.

Reply:
What I mean is nesting promise chains that don't need to be nested.

Reply:
@SimonSchick I'm not sure I totally follow what you're suggesting. I'm used to avoiding promise nesting all-together, particularly through the use of async/await. Perhaps you could provide an example so that I can get this patched up?

Reply:
It seems you already did that in your force push, didn't catch that :)

Reply:
👌